### PR TITLE
feature/20846 upgrade webpack

### PR DESCRIPTION
**Pull request by cbuchwald**
Updated packages and modified build scripts.

How to test:

1. npm run build
2. Test if everything works as before
3. Open two terminals with "npm run cypress:serve" and "npm run cypress:open" to test if everything still works

### CODE REVIEW

**Review by EgleHelms — APPROVED** [review]
Spec files are running well. Crated webchat.js file usable.

**Comment by pedily** [comment]
Why did you switch to individual webpack configuration files per service?
If we want to keep it that way, I would suggest to move them into a subfolder (like we have for Dockerfiles in build-utils) and change the naming to `webpack.plugin-name.js` and `webpack.plugin-name.legacy.js`

**Comment by pedily** [comment]
the legacy build outputs contain ES6 features like arrow functions and do not run in IE11

**Review by pedily — CHANGES_REQUESTED** [review]

File: package.json [hunk 1]
```diff
@@ -11,19 +11,19 @@
...
-    "rating": "webpack --config webpack.production-plugin.js --entry ./src/plugins/rating/index.ts --output dist/rating.webchat-plugin.js",
-    "speech-input-legacy": "webpack --config webpack.production-plugin.legacy.js --entry ./src/plugins/speech-input/index.tsx --output dist/speech-input.webchat-plugin.legacy.js",
-    "speech-input": "webpack --config webpack.production-plugin.js --entry ./src/plugins/speech-input/index.tsx --output dist/speech-input.webchat-plugin.js",
-    "speech-output-legacy": "webpack --config webpack.production-plugin.legacy.js --entry ./src/plugins/speech-output/index.tsx --output dist/speech-output.webchat-plugin.legacy.js",
-    "speech-output": "webpack --config webpack.production-plugin.js --entry ./src/plugins/speech-output/index.tsx --output dist/speech-output.webchat-plugin.js",
-    "webchat-legacy": "webpack --config webpack.production.legacy.js --output dist/webchat.legacy.js",
+    "date-picker-legacy": "webpack --config webpack.production-plugin.date-picker.legacy.js --entry ./src/plugins/date-picker/index.tsx",
+    "date-picker": "webpack --config webpack.production-plugin.date-picker.js --entry ./src/plugins/date-picker/index.tsx",
```
[the remark this answers is not in the document]
Comment (cbuchwald) [reply package.json:17]: Ok, will move it

**Comment by cbuchwald** [comment]
> Why did you switch to individual webpack configuration files per service? If we want to keep it that way, I would suggest to move them into a subfolder (like we have for Dockerfiles in build-utils) and change the naming to `webpack.plugin-name.js` and `webpack.plugin-name.legacy.js`

With Webpack 5 we no longer have the --output flag, therefore we need to have two files for each plugin and unfortunately it's not possible to move them to another folder, I already tried.

**Comment by cbuchwald** [comment]
> the legacy build outputs contain ES6 features like arrow functions and do not run in IE11

Thanks, that's a good catch, will see how I can fix this!

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (cbuchwald) [reply package.json]: moved

**Comment by pedily** [comment]
i checked compatibility for the legacy build with all the browsers we mention on our github page and it works

**Review by pedily — APPROVED** [review]

**Pull request merged**